tests/test_model_system.py

```diff
@@ -439,3 +442,242 @@ def test_normalize(self):
         assert np.isclose(model_system.elemental_composition[0].atomic_fraction, 2 / 3)
         assert model_system.elemental_composition[1].element == 'O'
         assert np.isclose(model_system.elemental_composition[1].atomic_fraction, 1 / 3)
+
+
+    @pytest.mark.parametrize(
+        'mol_label_list, n_mol_list, atom_labels_list, composition_formula_list',
+        [
+            (
+                ['H20'],
+                [3],
+                [['H', 'O', 'O']],
+                ['group_H20(1)', 'H20(3)', 'H(1)O(2)', 'H(1)O(2)', 'H(1)O(2)']
+            ), # pure system
+            (
+                ['H20', 'Methane'],
+                [5, 2],
+                [['H', 'O', 'O'], ['C', 'H', 'H', 'H', 'H']],
+                ['group_H20(1)group_Methane(1)', 'H20(5)', 'H(1)O(2)', 'H(1)O(2)', 'H(1)O(2)', 'H(1)O(2)', 'H(1)O(2)', 'Methane(2)', 'C(1)H(4)', 'C(1)H(4)']
+            ), # binary mixture
+        ],
+    )
+    def test_system_hierarchy_for_molecules(
```

Comment:
I think this test function is duplicated, you can delete it as the next one is the good one